wots.go

```diff
@@ -43,22 +46,38 @@ type Scheme struct {
 //
 // The hash function output size must have minimum 16 and maximum 128 bytes,
 // otherwise GenerateKeyPair method will always return error.
-func NewScheme(h func() hash.Hash, rand io.Reader) *Scheme {
+//
+// This variant of the function supports separate hash functions: (1) for
```

Comment:
Comments should start with FuncName. In general, Go std library usually uses phrasing "X2 is like X but...":

```
NewScheme2 is like NewScheme but accepts separate hash functions for (1) chaining and (2) for message hashing and final hashing into public key.

The requirement for the first function is to be PRF, while the later needs to be a collision resistant hash function.
```

Also, I think the order mentioned here is wrong :)